### PR TITLE
Add useIsGrouped hook, tests and use in inbound and outbounds

### LINKED ISSUE #684: Is grouped toggle is not persisted to local storage on inbound shipments

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #684 

- I made a hook `useIsGrouped` so I could consolidate all the local storage syncing together from the outbound component
- Added a couple tests and then off road moved the tests around into different describe scopes

### CODE REVIEW

File: packages/common/src/ui/layout/tables/context/TableContext.test.tsx [hunk 1]
```diff
@@ -1,11 +1,120 @@
...
 import { renderHook } from '@testing-library/react-hooks';
-import { createTableStore } from './TableContext';
+import { TableProvider, createTableStore } from './TableContext';
+import { LocalStorage } from '../../../../localStorage';
 
-describe('TableContext', () => {
-  const useStore = createTableStore();
+const useStore = createTableStore();
 
+describe('TableContext - expanding rows', () => {
```
Comment (joshxg) [inline packages/common/src/ui/layout/tables/context/TableContext.test.tsx:10]: Moved these expanding ones into their own describe block

File: packages/common/src/ui/layout/tables/context/TableContext.test.tsx [hunk 2]
```diff
@@ -139,109 +248,124 @@ describe('TableContext', () => {
     expect(result.current.rowState['c']?.isSelected).toBe(false);
     expect(result.current.numberSelected).toBe(0);
   });
+});
 
-  it('sets a single row as expanded once called', () => {
-    const { result } = renderHook<unknown, TableStore>(useStore);
+describe("TableContext - grouping rows", () => {
```
Comment (joshxg) [inline packages/common/src/ui/layout/tables/context/TableContext.test.tsx:253]: added these grouping row tests

File: packages/common/src/ui/layout/tables/context/TableContext.test.tsx [hunk 3]
```diff
@@ -139,109 +248,124 @@ describe('TableContext', () => {
...
+        expect(secondResult?.isGrouped).toBe(true);
+      }
 
-  it('sets all active rows as not expanded when expand all is called twice', () => {
-    const { result } = renderHook<unknown, TableStore>(useStore);
+      expect.assertions(2);
+    });
+})
+
+describe('useIsGrouped', () => {
```
Comment (joshxg) [inline packages/common/src/ui/layout/tables/context/TableContext.test.tsx:300]: and added these useIsGrouped tests

File: packages/common/src/ui/layout/tables/context/TableContext.test.tsx [hunk 4]
```diff
@@ -1,11 +1,120 @@
...
+
+    act(() => {
+      setActiveRows(['a', 'b', 'c']);
+    });
+
+    expect(result.current.numberExpanded).toBe(0);
+  });
+});
+
+describe('TableContext - setting selected rows', () => {
```
Comment (joshxg) [inline packages/common/src/ui/layout/tables/context/TableContext.test.tsx:117]: just all the selected row tests under this describe

File: packages/common/src/ui/layout/tables/context/TableContext.ts [hunk 5]
```diff
@@ -261,3 +261,30 @@ export const useDisabled = (rowId: string): UseDisabledControl => {
 
   return useTableStore(selector, equalityFn);
 };
+
+interface IsGroupedControl {
+  isGrouped: boolean;
+  toggleIsGrouped: () => void;
+}
+
+export const useIsGrouped = (key: keyof GroupByItem): IsGroupedControl => {
```
Comment (mark-prins) [inline packages/common/src/ui/layout/tables/context/TableContext.ts:270]: thanks - this is much better. good to have it all together

**Review by mark-prins — APPROVED** [review]
😁  lovely!

**Pull request merged**